edx_proctoring/api.py

```diff
@@ -1753,14 +1753,17 @@ def _get_proctored_exam_context(exam, attempt, user_id, course_id, is_practice_e
         'provider_name': provider.verbose_name,
     }
     if attempt:
-        provider_attempt = provider.get_attempt(attempt)
-        download_url = provider_attempt.get('download_url', None) or provider.get_software_download_url()
+        context['exam_code'] = attempt['attempt_code']
+        if attempt['status'] in (ProctoredExamStudentAttemptStatus.created,
+                                 ProctoredExamStudentAttemptStatus.download_software_clicked):
+            # since this may make an http request, let's not include it on every page
```

Comment:
Does this occur only for REST-based backends?

Reply:
Yes, non-REST backends don't make any request. `get_attempt` just returns the attempt passed in, unchanged.